### PR TITLE
feat: sign up

### LINKED ISSUE #341: Add option for user to register
If a user does not have an account, he should be prompted to create an account instead

### PR DESCRIPTION
**Pull request by mejariamol**
to be merged after #364 

- converted login activity into fragment
- created new module auth to contain login and signup
- added sign up feature

fixes #341 

screenshot_2017-07-27-11-33-34

### CODE REVIEW

**Comment by mejariamol** [comment]
@iamareebjamal
getting unsupported media type 415 error. please check the code.

File: app/src/main/java/org/fossasia/openevent/app/common/data/network/EventService.java [hunk 1]
```diff
@@ -18,6 +19,9 @@
 
 public interface EventService {
 
+    @POST("users")
+    Observable<User> signUp(@Body SignUp signUp);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/data/network/EventService.java]: This should be user object, not another SignUp object

**Comment by iamareebjamal** [comment]
The logs show what request body is being sent to the server, please check that. You can also use stetho

File: app/src/main/java/org/fossasia/openevent/app/common/data/network/EventService.java (hunk 1 above)
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/common/data/network/EventService.java]: but user doesn't have a field `password`

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/common/data/network/EventService.java]: @iamareebjamal the error showing is 
Content-Type header must be application/vnd.api+json

**Comment by mejariamol** [comment]
@iamareebjamal please review

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -14,8 +14,9 @@
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
         <activity
-            android:name=".module.login.LoginActivity"
-            android:theme="@style/AppTheme.NoActionBar">
+            android:name=".module.home.HomeActivity"
```
Comment (iamareebjamal) [inline app/src/main/AndroidManifest.xml]: Make this AuthActivity

File: app/src/main/java/org/fossasia/openevent/app/module/home/HomeActivity.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import org.fossasia.openevent.app.R;
+import org.fossasia.openevent.app.common.data.contract.IUtilModel;
+import org.fossasia.openevent.app.module.home.login.LoginFragment;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class HomeActivity extends AppCompatActivity{
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/module/home/HomeActivity.java]: Add space after AppCompatActivity

File: app/src/main/java/org/fossasia/openevent/app/module/home/signup/SignUpFragment.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    }
+
+    private void hideKeyBoard() {
+        View view = getActivity().getCurrentFocus();
+        if (view != null) {
+            InputMethodManager manager = (InputMethodManager)getActivity().getSystemService(Context.INPUT_METHOD_SERVICE);
+            view.clearFocus();
+            manager.hideSoftInputFromWindow(view.getWindowToken(), 0);
+        }
+    }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/module/home/signup/SignUpFragment.java]: If this is used in many places, this should be moved to ViewUtils

**Comment by iamareebjamal** [comment]
Also, add an option to set name, not only email and password. The person can leave it blank if he wants to

**Comment by mejariamol** [comment]
>keep the color tone consistent with the app

@iamareebjamal I didn't get this.

**Comment by mejariamol** [comment]
@iamareebjamal please review.

File: app/src/main/java/org/fossasia/openevent/app/common/data/contract/ILoginModel.java [hunk 5]
```diff
@@ -1,11 +1,16 @@
...
+import org.fossasia.openevent.app.common.data.models.User;
+
 import io.reactivex.Completable;
+import io.reactivex.Observable;
 
 public interface ILoginModel {
 
     Completable login(String username, String password);
 
+    Observable<User> signUp(String username, String password, String firstName, String lastName);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/data/contract/ILoginModel.java]: This should be a user object

File: app/src/main/java/org/fossasia/openevent/app/common/data/models/User.java [hunk 6]
```diff
@@ -51,4 +54,11 @@ public User() {}
     public User(String email) {
         this.email = email;
     }
+
+    public User(String email, String password, String firstName, String lastName) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/data/models/User.java]: Use lombok @Builder instead

**Comment by mejariamol** [comment]
@iamareebjamal please review.

**Comment by mariobehling** [comment]
Please resolve merge conflicts.

**Comment by mejariamol** [comment]
@iamareebjamal please do test on a device before review. I am moving to my college from mumbai to varanasi. So won't be available for two days.

**Pull request merged**